.github/workflows/integration_test_secrets.yaml

```diff
@@ -0,0 +1,77 @@
+name: Integration tests with secrets
+
+on:
+  workflow_call:
+
+env:
+  REGISTRY: ghcr.io
+  OWNER: ${{ github.repository_owner }}
+
+jobs:
+  get-runner-image:
+    name: Get runner image
+    uses: canonical/operator-workflows/.github/workflows/get_runner_image.yaml@main
+  build-images:
+    uses: canonical/operator-workflows/.github/workflows/build_images.yaml@main
+    needs: get-runner-image
+    with:
+      owner: ${{ github.repository_owner }}
+      registry: ghcr.io
+      runs-on: ${{ needs.get-runner-image.outputs.runs-on }}
+      trivy-image-config: trivy.yaml
+  integration-test-with-secrets:
+    runs-on: ubuntu-latest
+    needs: build-images
+    strategy:
+      fail-fast: false
+      matrix:
+        modules: [test_core]
+        args:
+          - name: num_units=3, db_from_relation
+            value: "--num-units=3"
+          - name: num_units=1, db_from_relation
+            value: "--num-units=1"
```

Comment:
Do we need to test with 1 and 3 units ?
I mean, is there a difference in the way secrets are managed between one and multiple units ?

Reply:
AFAIK, none. But as part of this PR I'm only adapting what we already had. I lack context to change this but I see the same issue in some of the other integration tests.